plugin/evm/atomic/state/atomic_repository.go

```diff
@@ -30,27 +30,17 @@ var (
 	atomicTxIDDBPrefix         = []byte("atomicTxDB")
 	atomicHeightTxDBPrefix     = []byte("atomicHeightTxDB")
 	atomicRepoMetadataDBPrefix = []byte("atomicRepoMetadataDB")
-	maxIndexedHeightKey        = []byte("maxIndexedAtomicTxHeight")
+	atomicTrieDBPrefix         = []byte("atomicTrieDB")
+	atomicTrieMetaDBPrefix     = []byte("atomicTrieMetaDB")
 
+	appliedSharedMemoryCursorKey = []byte("atomicTrieLastAppliedToSharedMemory")
+	maxIndexedHeightKey          = []byte("maxIndexedAtomicTxHeight")
 	// Historically used to track the completion of a migration
 	// bonusBlocksRepairedKey     = []byte("bonusBlocksRepaired")
 )
 
-// AtomicTxRepository defines an entity that manages storage and indexing of
-// atomic transactions
-type AtomicTxRepository interface {
-	GetIndexHeight() (uint64, error)
-	GetByTxID(txID ids.ID) (*atomic.Tx, uint64, error)
-	GetByHeight(height uint64) ([]*atomic.Tx, error)
-	Write(height uint64, txs []*atomic.Tx) error
-	WriteBonus(height uint64, txs []*atomic.Tx) error
-
-	IterateByHeight(start uint64) database.Iterator
-	Codec() codec.Manager
-}
-
-// atomicTxRepository is a prefixdb implementation of the AtomicTxRepository interface
-type atomicTxRepository struct {
+// AtomicRepository manages the database interactions for atomic operations.
+type AtomicRepository struct {
```

Comment:
Should we remove the `Atomic` prefix from the types as well? I could probably go either way here... They are inside of the `atomic` folder, but the actual package name is `state`...

It seems like we alias `state` to `atomicstate` a lot... So I'd probably prefer removing the `Atomic` prefix on the types here.

Reply:
https://github.com/ava-labs/coreth/pull/960#discussion_r2096132556